src/wiseagents/llm/openai_API_wise_agent_LLM.py

```diff
@@ -8,7 +8,13 @@ class OpenaiAPIWiseAgentLLM(WiseAgentRemoteLLM):
     client = None
     yaml_tag = u'!OpenaiAPIWiseAgentLLM'    
     
-    
+    def __new__(cls, *args, **kwargs):
+        '''Create a new instance of the class.'''
+        obj = super().__new__(cls)
+        obj._api_key = "sk-no-key-required"
```

Comment:
Should the default value for remote_address be set here too?

Reply:
I guess if someone supplies a yaml which doesn't have enough keys, those would cause the same problem. I'll add it and the other ones

Reply:
Right.

Reply:
Another thing I was thinking, is perhaps, since it uses the __dict__() method to set the values, if we could intercept that and make sure everything is set. But that feels more like relying on an implementation detail, `__new__()` seems more the Python contract

Reply:
Perhaps we need another validation step? Some ideas in https://stackoverflow.com/questions/3262569/validating-a-yaml-document-in-python

Reply:
Re: intercepting to make sure everything is set - that sounds similar to what we were doing in from_yaml, just getting the values directly and setting them if needed. But the `__new__()` approach does seem simpler.

Validation would probably be good to have in general. But for the things that have default values and don't have to be specified, we'd still need something like `__new__()`.

Reply:
I won't do this here since it is a bigger task, but I think a good approach could be:

* Add the `__new__()` methods like I am doing
* Add `from_yaml()` methods which delegate to the default pyyaml implementation (via some kind of super call or whatever the syntax would be). Then we can check everything is set properly